<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="John Powers" />

<meta name="date" content="2022-02-23" />

<title>Schiedea globosa scent - 2018</title>

<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libs/navigation-1.1/tabsets.js"></script>
<script src="libs/navigation-1.1/codefolding.js"></script>
<link href="libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Schiedea globosa scent - 2018</h1>
<h4 class="author"><a href="http://sites.uci.edu/powers/">John Powers</a></h4>
<h4 class="date">2022-02-23</h4>

</div>


<style type="text/css">
.main-container { max-width: 1000px; margin-left: 0; margin-right: auto; }
img{ max-width:200%; height: auto; }
td, th { padding : 6px }
</style>
<div id="inventory-of-all-years" class="section level1">
<h1>Inventory of all years</h1>
<pre class="r"><code>#Samples
#svglob %&gt;% count(DN,Year) %&gt;% pivot_wider(names_from=Year, values_from=n) %&gt;% 
#  kable(caption=&quot;Samples by year and time&quot;)
#svglob %&gt;% count(Sex, DN,Year) %&gt;% pivot_wider(names_from=Year, values_from=n, values_fill=0) %&gt;%
#  kable(caption=&quot;Samples by year, sex, and time&quot;)
sample_tally &lt;- svglob %&gt;% filter((Year==&quot;2018&quot; &amp; DN==&quot;Day&quot;) | (Year==&quot;2019&quot; &amp; DN==&quot;Night&quot;)) %&gt;% 
  count(Year, Island, Population, Sex) %&gt;% 
  pivot_wider(names_from=Sex, names_sort=T, values_from=n, values_fill=0)

svglob.unique &lt;- svglob %&gt;% 
  dplyr::select(Species, Island, Population, Family, Plant, Sex, Cutting, Year) %&gt;% 
  distinct() %&gt;% #drop DN and Sample and date to collapse day/night pairs and resamples #drop Cutting and PlantDate to get unique genotypes
  arrange(Species, Island, Population, Plant, Sex, Year)
#svglob.unique %&gt;% kable(caption=&quot;Plants in this study&quot;)

svglob.unique %&gt;% count(Population, Plant, Sex, Cutting) %&gt;% count(n, name=&quot;years_measured&quot;) %&gt;% 
  kable(caption=&quot;How many plants were measured in both years?&quot;)</code></pre>
<table>
<caption>How many plants were measured in both years?</caption>
<thead>
<tr class="header">
<th align="right">n</th>
<th align="right">years_measured</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="right">64</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">8</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#Genotypes
svglob.geno &lt;- svglob %&gt;% 
  dplyr::select(Species, Island, Population, Plant, Sex, Year) %&gt;% 
  distinct() %&gt;% #drop DN and Sample and date to collapse day/night pairs and resamples #drop Cutting and PlantDate to get unique genotypes
  arrange(Species, Island, Population, Plant, Sex, Year)
#with(svglob.geno, table(paste(Island, Population), Sex, Year))
#svglob.geno %&gt;% count(Year) %&gt;% kable(caption = &quot;Genotypes by year&quot;)
genotype_tally &lt;- svglob.geno %&gt;%  
  count(Year, Island, Population, Sex) %&gt;% 
  pivot_wider(names_from=Sex, names_sort=T, values_from=n, values_fill=0)

with(svglob, table(SampleDate, Sex)) %&gt;% chisq.test() %&gt;% tidy() %&gt;% 
  kable(caption = &quot;Selection of plants of each sex did not depend on date&quot;)</code></pre>
<table>
<caption>Selection of plants of each sex did not depend on date</caption>
<thead>
<tr class="header">
<th align="right">statistic</th>
<th align="right">p.value</th>
<th align="right">parameter</th>
<th align="left">method</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">50.92</td>
<td align="right">0.3594</td>
<td align="right">48</td>
<td align="left">Pearson’s Chi-squared test</td>
</tr>
</tbody>
</table>
<div id="appendix-s1-and-s3-inventory" class="section level2">
<h2>Appendix S1 and S3: Inventory</h2>
<pre class="r"><code>bind_rows(sample=sample_tally, genotype=genotype_tally, .id=&quot;count&quot;) %&gt;% 
  pivot_wider(names_from = &quot;count&quot;, values_from = c(&quot;F&quot;, &quot;H&quot;, &quot;M&quot;)) %&gt;% 
  kable(caption = paste0(&quot;Samples and genotypes by year, population, and sex&quot;))</code></pre>
<table style="width:100%;">
<caption>Samples and genotypes by year, population, and sex</caption>
<colgroup>
<col width="5%" />
<col width="10%" />
<col width="12%" />
<col width="10%" />
<col width="12%" />
<col width="10%" />
<col width="12%" />
<col width="10%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Year</th>
<th align="left">Island</th>
<th align="left">Population</th>
<th align="right">F_sample</th>
<th align="right">F_genotype</th>
<th align="right">H_sample</th>
<th align="right">H_genotype</th>
<th align="right">M_sample</th>
<th align="right">M_genotype</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">2018</td>
<td align="left">O’ahu</td>
<td align="left">13886</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">6</td>
<td align="right">6</td>
</tr>
<tr class="even">
<td align="left">2018</td>
<td align="left">O’ahu</td>
<td align="left">13885</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">2018</td>
<td align="left">O’ahu</td>
<td align="left">964</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">2018</td>
<td align="left">O’ahu</td>
<td align="left">844</td>
<td align="right">4</td>
<td align="right">5</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">2018</td>
<td align="left">O’ahu</td>
<td align="left">906</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">2018</td>
<td align="left">Moloka’i</td>
<td align="left">10228</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">2018</td>
<td align="left">Moloka’i</td>
<td align="left">102HU</td>
<td align="right">4</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">2018</td>
<td align="left">Maui</td>
<td align="left">905</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">2018</td>
<td align="left">Hawai’i</td>
<td align="left">15191</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">2019</td>
<td align="left">O’ahu</td>
<td align="left">13886</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">2019</td>
<td align="left">O’ahu</td>
<td align="left">13885</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">2019</td>
<td align="left">O’ahu</td>
<td align="left">964</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">2019</td>
<td align="left">O’ahu</td>
<td align="left">844</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">2019</td>
<td align="left">O’ahu</td>
<td align="left">906</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">2019</td>
<td align="left">Moloka’i</td>
<td align="left">10228</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">2019</td>
<td align="left">Moloka’i</td>
<td align="left">102HU</td>
<td align="right">4</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">2019</td>
<td align="left">Maui</td>
<td align="left">951</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">2019</td>
<td align="left">Maui</td>
<td align="left">905</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">3</td>
</tr>
<tr class="odd">
<td align="left">2019</td>
<td align="left">Maui</td>
<td align="left">850</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">2019</td>
<td align="left">Hawai’i</td>
<td align="left">11221</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">2019</td>
<td align="left">Hawai’i</td>
<td align="left">15191</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="scent-vs.-infloresence-age" class="section level1">
<h1>Scent vs. infloresence age</h1>
<pre class="r"><code>#Scent composition ~ open flowers
capscale(decostand(vol.glob, &quot;tot&quot;) ~ Year+DN+Sex+Flrs, data=svglob, distance=&quot;bray&quot;, metaMDSdist = F) %&gt;% 
  anova(by=&quot;term&quot;) %&gt;% tidy() %&gt;% kable(caption = &quot;CAP of scent composition vs. number of flowers&quot;)</code></pre>
<table>
<caption>CAP of scent composition vs. number of flowers</caption>
<thead>
<tr class="header">
<th align="left">term</th>
<th align="right">df</th>
<th align="right">SumOfSqs</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Year</td>
<td align="right">1</td>
<td align="right">3.3599</td>
<td align="right">17.762</td>
<td align="right">0.001</td>
</tr>
<tr class="even">
<td align="left">DN</td>
<td align="right">1</td>
<td align="right">1.0637</td>
<td align="right">5.623</td>
<td align="right">0.001</td>
</tr>
<tr class="odd">
<td align="left">Sex</td>
<td align="right">2</td>
<td align="right">0.6468</td>
<td align="right">1.710</td>
<td align="right">0.014</td>
</tr>
<tr class="even">
<td align="left">Flrs</td>
<td align="right">1</td>
<td align="right">0.2226</td>
<td align="right">1.177</td>
<td align="right">0.251</td>
</tr>
<tr class="odd">
<td align="left">Residual</td>
<td align="right">128</td>
<td align="right">24.2129</td>
<td align="right"></td>
<td align="right"></td>
</tr>
</tbody>
</table>
<pre class="r"><code>  #Scent compostion ~ inflo age
svglob$Age &lt;- with(svglob, Closed/(Closed+Flrs))
capscale(decostand(vol.glob, &quot;tot&quot;) ~ Year+DN+Sex+Age, data=svglob, distance=&quot;bray&quot;, metaMDSdist = F) %&gt;% 
  anova(by=&quot;term&quot;) %&gt;% tidy() %&gt;% kable(caption = &quot;CAP of scent composition vs. inflo age (closed/total flowers)&quot;)</code></pre>
<table>
<caption>CAP of scent composition vs. inflo age (closed/total flowers)</caption>
<thead>
<tr class="header">
<th align="left">term</th>
<th align="right">df</th>
<th align="right">SumOfSqs</th>
<th align="right">statistic</th>
<th align="right">p.value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Year</td>
<td align="right">1</td>
<td align="right">3.3599</td>
<td align="right">17.7023</td>
<td align="right">0.001</td>
</tr>
<tr class="even">
<td align="left">DN</td>
<td align="right">1</td>
<td align="right">1.0637</td>
<td align="right">5.6042</td>
<td align="right">0.001</td>
</tr>
<tr class="odd">
<td align="left">Sex</td>
<td align="right">2</td>
<td align="right">0.6468</td>
<td align="right">1.7038</td>
<td align="right">0.016</td>
</tr>
<tr class="even">
<td align="left">Age</td>
<td align="right">1</td>
<td align="right">0.1413</td>
<td align="right">0.7447</td>
<td align="right">0.749</td>
</tr>
<tr class="odd">
<td align="left">Residual</td>
<td align="right">128</td>
<td align="right">24.2941</td>
<td align="right"></td>
<td align="right"></td>
</tr>
</tbody>
</table>
<pre class="r"><code># Restrict to `r params$year` samples only
if(params$year==2018) dn &lt;- &quot;Day&quot; else dn &lt;- c(&quot;Day&quot;,&quot;Night&quot;)
vol.glob &lt;-   vol.glob[svglob$Year == as.character(params$year) &amp; svglob$DN %in% dn,]
fvol.glob &lt;- fvol.glob[svglob$Year == as.character(params$year) &amp; svglob$DN %in% dn,]
svglob &lt;-       svglob[svglob$Year == as.character(params$year) &amp; svglob$DN %in% dn,]

#populations in order of island
svglob$Population &lt;- factor(svglob$Population, levels=glob.pops.WtoE)
glob.pops.pal &lt;- setNames(c(brewer.pal(5,&quot;Purples&quot;)[c(3,3,3,4,5)], brewer.pal(4,&quot;Blues&quot;)[3:4], brewer.pal(5,&quot;Greens&quot;)[3:5], brewer.pal(4,&quot;Oranges&quot;)[3:4]), glob.pops.WtoE)#changed O&#39;ahu 13886 (first one) to 3
svglob$IslandPop &lt;- factor(with(svglob, paste(Island, Population)), levels=paste(svglob$Island[match(glob.pops.WtoE, svglob$Population)], glob.pops.WtoE))

svglob$IslandPopL &lt;- svglob$IslandPop# L for lumped
levels(svglob$IslandPopL)[1:3] &lt;- &quot;O&#39;ahu Ko&#39;olau&quot;

#For genetics, missing two Wood pops
svglob$PopulationL &lt;- svglob$Population# L for lumped
levels(svglob$PopulationL)[1:2] &lt;- &quot;964&quot; #lump Kawelo 964 and Wood 13885 - Kanehoalani # lump Kaaawa 13886 into Kanehoalani
glob.popsL.pal &lt;- glob.pops.pal[-c(1:2)]</code></pre>
<pre class="r"><code>EZ &lt;- sort(fvol.glob$`(E)-3-methylbutyraldoxime`/fvol.glob$`(Z)-3-methylbutyraldoxime`)[-c(1,53:100)]
print(paste(&quot;E/Z emission ratio for3-methylbutyraldoxime:&quot;,format(mean(EZ), digits=2)))</code></pre>
</div>
<div id="appendix-s3-model-of-total-emissions" class="section level1">
<h1>Appendix S3: Model of total emissions</h1>
<pre class="r"><code>svglob$Genotype &lt;- with(svglob, factor(paste(Population, Plant, Sex)))

if(params$year == 2019) {
svglob$SexDN &lt;- with(svglob, factor(paste0(Sex, DN)))
SexDN.x &lt;- data.frame(SexDN=levels(svglob$SexDN), x=c(1,2,5,6,3,4))
ftot.mod &lt;-     lmer(fvoltot ~ SexDN +             (1|Genotype/Pair), data = svglob) 
ftot.mod.noH &lt;- lmer(fvoltot ~ DN*Sex+Population + (1|Genotype/Pair), data = svglob[svglob$Sex!=&quot;H&quot;,])# use this one since herm sample size is low.
ftot.mod.noH.anova &lt;- anova(ftot.mod.noH, type=3)

cld.mod &lt;- ftot.mod %&gt;% ref_grid %&gt;% emmeans(~ SexDN) %&gt;% summary 

cont &lt;- ftot.mod %&gt;%
multcomp::glht(linfct = multcomp::mcp(SexDN=c(&quot;FDay  - FNight == 0&quot;,
                                              &quot;MDay  - MNight == 0&quot;,
                                              &quot;HDay  - HNight == 0&quot;,
                                              &quot;FDay  - MDay == 0&quot;,
                                              &quot;FNight  - MNight == 0&quot;
                                              ))) %&gt;%
  summary(test=multcomp::adjusted(type=&quot;fdr&quot;)) %&gt;% tidy %&gt;% 
  separate(contrast, c(&quot;lvl1&quot;, &quot;lvl2&quot;), sep = &quot; - &quot;) %&gt;%
  mutate(signif = adj.p.value &lt; 0.05, 
         stars = signif.num(adj.p.value), 
         height = c(1,1,1,2,3), 
         increase=(-estimate/cld.mod[match(lvl1, cld.mod$SexDN),&quot;emmean&quot;]),
         x1 = SexDN.x[match(lvl1, SexDN.x$SexDN),&quot;x&quot;], 
         x2 = SexDN.x[match(lvl2, SexDN.x$SexDN),&quot;x&quot;])
#cont[,-c(4,9,11,13,14)] %&gt;% kable(digits=2, caption=&quot;Contrasts between total emissions for each sex and time&quot;) 
} else if(params$year == 2018) {
  ftot.mod &lt;-     lm(fvoltot ~ Sex,            data = svglob) 
  ftot.mod.noH &lt;- lm(fvoltot ~ Sex+Population, data = svglob[svglob$Sex!=&quot;H&quot;,])
  ftot.mod.noH.anova &lt;- anova(ftot.mod.noH)
  cld.mod &lt;- ftot.mod %&gt;% ref_grid %&gt;% emmeans(~ Sex) %&gt;% summary 
}
ftot.mod.noH.anova %&gt;% kable(caption=&quot;A. Model of total emissions (excluding hermaphrodites)&quot;)</code></pre>
<table>
<caption>A. Model of total emissions (excluding hermaphrodites)</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Df</th>
<th align="right">Sum Sq</th>
<th align="right">Mean Sq</th>
<th align="right">F value</th>
<th align="right">Pr(&gt;F)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Sex</td>
<td align="right">1</td>
<td align="right">3047</td>
<td align="right">3047.0</td>
<td align="right">1.9088</td>
<td align="right">0.1804</td>
</tr>
<tr class="even">
<td align="left">Population</td>
<td align="right">8</td>
<td align="right">6101</td>
<td align="right">762.6</td>
<td align="right">0.4777</td>
<td align="right">0.8593</td>
</tr>
<tr class="odd">
<td align="left">Residuals</td>
<td align="right">23</td>
<td align="right">36715</td>
<td align="right">1596.3</td>
<td align="right"></td>
<td align="right"></td>
</tr>
</tbody>
</table>
<pre class="r"><code>cld.mod %&gt;% kable(digits=0, caption = &quot;B. Estimated marginal means for total emissions&quot;)</code></pre>
<table>
<caption>B. Estimated marginal means for total emissions</caption>
<thead>
<tr class="header">
<th align="left">Sex</th>
<th align="right">emmean</th>
<th align="right">SE</th>
<th align="right">df</th>
<th align="right">lower.CL</th>
<th align="right">upper.CL</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">F</td>
<td align="right">35</td>
<td align="right">9</td>
<td align="right">33</td>
<td align="right">16</td>
<td align="right">53</td>
</tr>
<tr class="even">
<td align="left">H</td>
<td align="right">43</td>
<td align="right">21</td>
<td align="right">33</td>
<td align="right">0</td>
<td align="right">87</td>
</tr>
<tr class="odd">
<td align="left">M</td>
<td align="right">54</td>
<td align="right">9</td>
<td align="right">33</td>
<td align="right">35</td>
<td align="right">73</td>
</tr>
</tbody>
</table>
</div>
<div id="appendix-s3-total-emissions" class="section level1">
<h1>Appendix S3: Total emissions</h1>
<pre class="r"><code>if(params$year == 2019) {
  (ftot.plot &lt;- ggplot(svglob, aes(x=factor(paste(Sex, DN), levels=c(&quot;F Day&quot;,&quot;F Night&quot;,&quot;M Day&quot;,&quot;M Night&quot;,&quot;H Day&quot;,&quot;H Night&quot;)), 
                                   y = fvoltot, color=factor(Sex, levels=c(&quot;F&quot;,&quot;M&quot;,&quot;H&quot;)), 
                                   shape=factor(paste(Sex,DN), levels=c(&quot;F Day&quot;,&quot;F Night&quot;,&quot;M Day&quot;,&quot;M Night&quot;,&quot;H Day&quot;,&quot;H Night&quot;)))) + 
     geom_signif(y_position=15*cont$height+max(svglob$fvoltot)*1.01, xmin = cont$x1, xmax = cont$x2, 
                 annotations=paste0(&quot;+&quot;,round(cont$increase*100),&quot;%, P &quot;,ifelse(cont$adj.p.value &lt; 0.001, &quot;&lt; 0.001&quot;,paste(&quot;=&quot;,round(cont$adj.p.value,3)))), 
                 vjust=0, color=&quot;black&quot;, tip_length=0.01) +
     geom_path(aes(group=Pair))+
     geom_point(size=2) + 
     scale_y_continuous(&quot;Emission rate (ng/h/flower)&quot;, breaks=seq(0,200, by=50), minor_breaks=seq(0,225, by=25), limits=c(0,NA)) + 
     scale_shape_manual(values=c(1,16,5,18,0,15))+
     scale_color_brewer(palette=&quot;Set1&quot;)+
     scale_x_discrete(&quot;&quot;,labels=c(&quot;Female\nDay&quot;,&quot;Female\nNight&quot;,&quot;Male\nDay&quot;,&quot;Male\nNight&quot;, &quot;Herm.\nDay&quot;,&quot;Herm.\nNight&quot;))+
     guides(shape=F, color=F)+
     theme_minimal()+theme(panel.grid.major.x = element_blank(), axis.text=element_text(color=&quot;black&quot;)))
  
  ggsave(&quot;./output_cut/Fig2.pdf&quot;, ftot.plot, device=cairo_pdf, height=7, width=6)
  
} else if(params$year == 2018) {
  ftot.plot &lt;- ggplot(svglob, aes(x=factor(paste(Sex, DN), levels=c(&quot;F Day&quot;,&quot;M Day&quot;,&quot;H Day&quot;)), 
                                   y = fvoltot, color=factor(Sex, levels=c(&quot;F&quot;,&quot;M&quot;,&quot;H&quot;)), 
                                   shape=factor(paste(Sex,DN), levels=c(&quot;F Day&quot;,&quot;M Day&quot;,&quot;H Day&quot;)))) + 
     geom_boxplot(color=c(brewer.pal(3,&quot;Set1&quot;)[1:2],NA), outlier.shape=NA, fill=NA)+
     geom_point(size=2) + 
     scale_y_continuous(&quot;Emission rate (ng/h/flower)&quot;, breaks=seq(0,250, by=50), minor_breaks=seq(0,250, by=25), limits=c(0,NA)) + 
     scale_shape_manual(values=c(1,5,0))+
     scale_color_brewer(palette=&quot;Set1&quot;)+
     scale_x_discrete(&quot;&quot;,labels=c(&quot;Female\nDay&quot;,&quot;Male\nDay&quot;,&quot;Herm.\nDay&quot;))+
     guides(shape=F, color=F)+
     theme_minimal()+theme(panel.grid.major.x = element_blank(), axis.text=element_text(color=&quot;black&quot;))
  
  ggsave(&quot;./output_cut/AppendixS3_total.pdf&quot;, ftot.plot, device=cairo_pdf, height=5, width=3)
}
print(ftot.plot)</code></pre>
<p><img src="figures2018/total-1.svg" width="288" /></p>
</div>
<div id="appendix-s3-scent-composition-model" class="section level1">
<h1>Appendix S3: Scent composition model</h1>
<pre class="r"><code>if(params$year == 2019) {
  cap.glob.pop.prop &lt;- capscale(decostand(vol.glob[svglob$Sex!=&quot;H&quot;,], &quot;hellinger&quot;) ~ StartSunset*Sex+Population,
                                data=svglob[svglob$Sex!=&quot;H&quot;,], distance=&quot;bray&quot;, metaMDSdist = F) 
} else if(params$year == 2018) {
  cap.glob.pop.prop &lt;- capscale(decostand(vol.glob[svglob$Sex!=&quot;H&quot;,], &quot;hellinger&quot;) ~ Sex+Population, 
                                data=svglob[svglob$Sex!=&quot;H&quot;,], distance=&quot;bray&quot;, metaMDSdist = F) 
}

cap.glob.pop.prop</code></pre>
<pre><code>Call: capscale(formula = decostand(vol.glob[svglob$Sex != &quot;H&quot;, ],
&quot;hellinger&quot;) ~ Sex + Population, data = svglob[svglob$Sex != &quot;H&quot;, ],
distance = &quot;bray&quot;, metaMDSdist = F)

              Inertia Proportion Rank
Total          5.6729     1.0000     
Constrained    2.4249     0.4275    9
Unconstrained  3.7920     0.6684   19
Imaginary     -0.5440    -0.0959   13
Inertia is squared Bray distance 
Species scores projected from &#39;decostand&#39; &#39;vol.glob[svglob$Sex != &quot;H&quot;, ]&#39; &#39;hellinger&#39; 

Eigenvalues for constrained axes:
 CAP1  CAP2  CAP3  CAP4  CAP5  CAP6  CAP7  CAP8  CAP9 
1.012 0.513 0.304 0.213 0.156 0.103 0.056 0.051 0.016 

Eigenvalues for unconstrained axes:
 MDS1  MDS2  MDS3  MDS4  MDS5  MDS6  MDS7  MDS8 
0.843 0.679 0.480 0.403 0.353 0.252 0.163 0.132 
(Showing 8 of 19 unconstrained eigenvalues)</code></pre>
<pre class="r"><code>set.seed(1)
anova.cap.glob.pop.prop &lt;- anova(cap.glob.pop.prop, by=&quot;term&quot;)

anova.cap.df &lt;- anova.cap.glob.pop.prop %&gt;% as.data.frame %&gt;% rownames_to_column(&quot;term&quot;) %&gt;% 
  filter(term!=&quot;Total&quot;) %&gt;% mutate(R2 = SumOfSqs / sum(SumOfSqs), term=term %&gt;% 
                                     str_replace_all(c(&quot;StartSunset&quot;=&quot;Time&quot;, &quot;:&quot;=&quot; \u00D7 &quot;)) %&gt;% fct_reorder(R2))
anova.cap.df %&gt;% kable(caption = &quot;Permutation test of CAP model of relative emissions&quot;)</code></pre>
<table>
<caption>Permutation test of CAP model of relative emissions</caption>
<thead>
<tr class="header">
<th align="left">term</th>
<th align="right">Df</th>
<th align="right">SumOfSqs</th>
<th align="right">F</th>
<th align="right">Pr(&gt;F)</th>
<th align="right">R2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Sex</td>
<td align="right">1</td>
<td align="right">0.3287</td>
<td align="right">1.994</td>
<td align="right">0.031</td>
<td align="right">0.0529</td>
</tr>
<tr class="even">
<td align="left">Population</td>
<td align="right">8</td>
<td align="right">2.0962</td>
<td align="right">1.589</td>
<td align="right">0.005</td>
<td align="right">0.3372</td>
</tr>
<tr class="odd">
<td align="left">Residual</td>
<td align="right">23</td>
<td align="right">3.7920</td>
<td align="right"></td>
<td align="right"></td>
<td align="right">0.6100</td>
</tr>
</tbody>
</table>
</div>
<div id="appendix-s3-daynight-differences" class="section level1">
<h1>Appendix S3: Day/night differences</h1>
<pre class="r"><code>if(params$year == 2019) {
  (cap.glob &lt;- capscale(decostand(fvol.glob, &quot;hellinger&quot;) ~ Time+Sex, data=svglob, distance=&quot;bray&quot;, metaMDSdist = F))
  
  layout(matrix(c(1,2), 2, 1), heights=c(8,4.5))
  par(mar=c(5,4,0,0))
  plot(cap.glob, type=&quot;n&quot;)
  mtext(&quot;A&quot;,side=2, line=2.5,padj=-13, cex=2, las=2)
  legend(&quot;topright&quot;, c(&quot;Female&quot;,&quot;Herm&quot;,&quot;Male&quot;), cex =1.2, inset=c(0.01,0), bty=&quot;n&quot;, adj=c(0.5,2), text.col=brewer.pal(3, &quot;Set1&quot;)[c(1,3,2)])
  legend(&quot;topright&quot;, title=&quot;Day&quot;, c(&quot;&quot;,&quot;&quot;,&quot;&quot;), cex =1.2, pch = c(1,5,0), pt.cex=c(1.5,1.5,1.5), inset=c(0.15,0), bty=&quot;n&quot;, col=brewer.pal(3, &quot;Set1&quot;)[c(1,3,2)], pt.lwd=4, adj=c(0.5,0.5))
  legend(&quot;topright&quot;, c(&quot;&quot;,&quot;&quot;,&quot;&quot;), title=&quot;Night&quot;, cex =1.2, pch = c(16,18,15), pt.cex=c(1.5,2,1.5), inset=c(0,0),  bty=&quot;n&quot;, col=brewer.pal(3, &quot;Set1&quot;)[c(1,3,2)], pt.lwd=4)
  ordihull(cap.glob, with(svglob, paste(Species, Population, Plant, Cutting, Sex, lubridate::week(date))), col=alpha(&quot;grey60&quot;, 0.6), lwd=2)
  points(cap.glob, display=&quot;sites&quot;, col=brewer.pal(3, &quot;Set1&quot;)[c(1,3,2)][svglob$Sex],  
         pch=c(1,16,5,18,0,15)[(as.integer(svglob$Sex)*2 + as.integer(svglob$DN))-2], 
         cex=ifelse(svglob$Sex==&quot;H&quot; &amp; svglob$DN==&quot;Night&quot;,2,ifelse(svglob$Sex==&quot;F&quot; &amp; svglob$DN==&quot;Night&quot;,1.5,1.5)), lwd=4)
  text(cap.glob, display=&quot;cn&quot;, labels=c(&quot;Female&quot;, &quot;Herm.&quot;, &quot;Male&quot;, &quot;Time&quot;), font=2, arrow.mul=1.15, head.arrow=0.1, lwd=2)

  chem_sum_cutoff &lt;- 40
  par(mar=c(4,4,0,0))
  plot(cap.glob, type=&quot;n&quot;, xlim=c(-1.2,1), ylim=c(-0.6,0.6))
  mtext(&quot;B&quot;,side=2, line=2.5,padj=-6, cex=2, las=2)
  legend(&quot;bottomleft&quot; ,names(chemtype.pal), col=chemtype.pal, pt.cex=2, pch=19, bty=&quot;n&quot;)
  points(cap.glob, display=&quot;species&quot;, col=chemtype.pal[fvol.glob.mean$type], pch=ifelse(colSums(fvol.glob)&gt;chem_sum_cutoff, 19, NA), cex=0.6)
  text(cap.glob, display=&quot;cn&quot;, labels=c(&quot;Female&quot;, &quot;Herm.&quot;, &quot;Male&quot;,&quot;Time&quot;), font=2, arrow.mul=0.9, head.arrow=0.1, lwd=2)
  set.seed(17) #get seed by force below
  ordipointlabel(cap.glob, display=&quot;species&quot;, select=colSums(fvol.glob)&gt;chem_sum_cutoff,
                 col=chemtype.pal[fvol.glob.mean$type][colSums(fvol.glob)&gt;chem_sum_cutoff], font=2, xpd=T, pch=19, add=T, cex=0.8)
  dev.copy(cairo_pdf, filename=&quot;./output_cut/Fig3AB.pdf&quot;, width=8, height=8+4.5)
  dev.off()
  
} else if(params$year == 2018) {
  (cap.glob &lt;- capscale(decostand(fvol.glob, &quot;hellinger&quot;) ~ Sex, data=svglob, distance=&quot;bray&quot;, metaMDSdist = F))
  
  layout(matrix(c(1,2), 2, 1), heights=c(8,4.5))
  par(mar=c(4,4,1,1))
  plot(cap.glob, type=&quot;n&quot;)
  mtext(&quot;A&quot;,side=2, line=2.5,padj=-13, cex=2, las=2)
  points(cap.glob, display=&quot;sites&quot;, col=brewer.pal(3, &quot;Set1&quot;)[c(1,3,2)][svglob$Sex], pch=c(1,5,0)[as.integer(svglob$Sex)], cex=1.5, lwd=4)
  text(cap.glob, display=&quot;cn&quot;, labels=c(&quot;Female&quot;, &quot;Herm.&quot;, &quot;Male&quot;), font=2, col=brewer.pal(3, &quot;Set1&quot;)[c(1,3,2)])
  legend(&quot;topleft&quot;, &quot;Day&quot;, bty=&quot;n&quot;, cex=1.2, inset=c(-0.04, 0))
  
  chem_sum_cutoff &lt;- 40
  plot(cap.glob, type=&quot;n&quot;, xlim=c(-0.4,0.4), ylim=c(-0.4,0.4))
  mtext(&quot;B&quot;,side=2, line=2.5,padj=-6, cex=2, las=2)
  legend(&quot;bottomleft&quot; ,names(chemtype.pal), col=chemtype.pal, pt.cex=2, pch=19, bty=&quot;n&quot;)
  points(cap.glob, display=&quot;species&quot;, col=chemtype.pal[fvol.glob.mean$type], pch=ifelse(colSums(fvol.glob)&gt;chem_sum_cutoff, 19, NA), cex=0.6)
  set.seed(1)
  ordipointlabel(cap.glob, display=&quot;species&quot;, select=colSums(fvol.glob)&gt;chem_sum_cutoff,
                 col=chemtype.pal[fvol.glob.mean$type][colSums(fvol.glob)&gt;chem_sum_cutoff], font=2, xpd=T, pch=19, add=T, cex=0.8)
    text(cap.glob, display=&quot;cn&quot;, labels=c(&quot;Female&quot;, &quot;Herm.&quot;, &quot;Male&quot;), font=2)
  dev.copy(cairo_pdf, filename=&quot;./output_cut/AppendixS3_CAP_AB.pdf&quot;, width=8, height=8+4.5)
  dev.off()
}</code></pre>
<p><img src="figures2018/cap_DN_Sex-1.svg" width="768" /></p>
</div>
<div id="appendix-s3-population-differences" class="section level1">
<h1>Appendix S3: Population differences</h1>
<pre class="r"><code>svglob.D &lt;- svglob[svglob$DN==&quot;Day&quot;,]
svglob.D$Population &lt;- droplevels(svglob.D$Population)#not all pops in 2018 dataset
svglob.D$IslandPop &lt;- droplevels(svglob.D$IslandPop)
cap.glob.D &lt;- capscale(decostand(fvol.glob[svglob$DN==&quot;Day&quot;,], &quot;hellinger&quot;) ~ Population, data=svglob.D, distance=&quot;bray&quot;, metaMDSdist = F) 


if(params$year==2019) {
chem_sum_cutoff &lt;- 60

par(mar=c(5,4,0,0))
layout(matrix(c(1,2), 2, 1), heights=c(8,8))
plot(cap.glob.D, type=&quot;n&quot;, xlim=c(-1,1))
mtext(&quot;C&quot;,side=2, line=2.5,padj=-13, cex=2, las=2)
ordispider(cap.glob.D, svglob.D$PopulationL, display=&quot;sites&quot;, col=glob.popsL.pal)
points(cap.glob.D, display=&quot;sites&quot;, col=glob.popsL.pal[svglob.D$PopulationL], 
       pch=c(1,16,5,18,0,15)[(as.integer(svglob.D$Sex)*2 + as.integer(svglob.D$DN))-2], 
       cex=ifelse(svglob.D$Sex==&quot;H&quot; &amp; svglob.D$DN==&quot;Night&quot;,2,1.5), lwd=4)
points(cap.glob.D, display=&quot;species&quot;, col=alpha(&quot;black&quot;, 0.5), pch=ifelse(colSums(fvol.glob)&gt;chem_sum_cutoff, 17, NA), cex=0.8)
set.seed(2)
ordipointlabel(cap.glob.D, display=&quot;species&quot;, select=colSums(fvol.glob)&gt;chem_sum_cutoff, 
               col=alpha(&quot;black&quot;, 0.5), font=2, pch=19, add=T, cex=0.8) 
legend(&quot;bottomleft&quot;, levels(svglob.D$IslandPopL), text.col=glob.popsL.pal, cex =1.2, inset=c(-0.03,0), xpd=T, bty=&quot;n&quot;)
legend(&quot;bottomright&quot;, c(&quot;Female&quot;,&quot;Herm.&quot;,&quot;Male&quot;), cex =1.2, pch = c(1,5,0), pt.cex=1.5, inset=c(-0,0), xpd=T, bty=&quot;n&quot;, pt.lwd=4)
legend(&quot;topleft&quot;, &quot;Day&quot;, bty=&quot;n&quot;, cex=1.2, inset=c(-0.04, 0))

svglob.N &lt;- svglob[svglob$DN==&quot;Night&quot;,]
cap.glob.N &lt;- capscale(decostand(fvol.glob[svglob$DN==&quot;Night&quot;,], &quot;hellinger&quot;) ~ Population, data=svglob.N, distance=&quot;bray&quot;, metaMDSdist = F) 
par(mar=c(4,4,0,0))
plot(cap.glob.N, type=&quot;n&quot;, xlim=c(-1.4,0.8))
mtext(&quot;D&quot;,side=2, line=2.5,padj=-13, cex=2, las=2)
ordispider(cap.glob.N, svglob.N$PopulationL, display=&quot;sites&quot;, col=glob.popsL.pal)
points(cap.glob.N, display=&quot;sites&quot;, col=glob.popsL.pal[svglob.N$PopulationL], pch=c(1,16,5,18,0,15)[(as.integer(svglob.N$Sex)*2 + as.integer(svglob.N$DN))-2], cex=ifelse(svglob.N$Sex==&quot;H&quot; &amp; svglob.N$DN==&quot;Night&quot;,2,1.5), lwd=4)
points(cap.glob.N, display=&quot;species&quot;, col=alpha(&quot;black&quot;,0.5), pch=ifelse(colSums(fvol.glob)&gt;chem_sum_cutoff, 17, NA), cex=0.8)#col=chemtype.pal[fvol.glob.mean$type]
set.seed(2)
ordipointlabel(cap.glob.N, display=&quot;species&quot;, select=colSums(fvol.glob)&gt;chem_sum_cutoff, col=alpha(&quot;black&quot;,0.5), font=2, pch=19, add=T, cex=0.75)
legend(&quot;bottomright&quot;, c(&quot;Female&quot;,&quot;Herm.&quot;,&quot;Male&quot;), cex =1.2, pch = c(1,16,5,18,0,15)[c(2,4,6)], pt.cex=c(1.5,1.5,1.5,2,1.5,1.5,1.5)[c(2,4,6)], inset=c(-0,0), xpd=T, bty=&quot;n&quot;, pt.lwd=4)
legend(&quot;topleft&quot;, &quot;Night&quot;, bty=&quot;n&quot;, cex=1.2, inset=c(-0.04, 0))
dev.copy(cairo_pdf, filename=&quot;./output_cut/Fig3CD.pdf&quot;, width=8, height=16)
dev.off()

} else if(params$year==2018) {
chem_sum_cutoff &lt;- 60

par(mar=c(4,4,0,0))
plot(cap.glob.D, type=&quot;n&quot;)
mtext(&quot;C&quot;,side=2, line=2.5,padj=-13, cex=2, las=2)
ordispider(cap.glob.D, svglob.D$PopulationL, display=&quot;sites&quot;, col=glob.popsL.pal)
points(cap.glob.D, display=&quot;sites&quot;, col=glob.popsL.pal[svglob.D$PopulationL], 
       pch=c(1,5,0)[as.integer(svglob.D$Sex)], cex=1.5, lwd=4)
points(cap.glob.D, display=&quot;species&quot;, col=alpha(&quot;black&quot;, 0.5), pch=ifelse(colSums(fvol.glob)&gt;chem_sum_cutoff, 17, NA), cex=0.8)
set.seed(2)
ordipointlabel(cap.glob.D, display=&quot;species&quot;, select=colSums(fvol.glob)&gt;chem_sum_cutoff, 
               col=alpha(&quot;black&quot;, 0.5), font=2, pch=19, add=T, cex=0.8) 
legend(&quot;bottomleft&quot;, levels(svglob.D$IslandPopL)[table(svglob.D$IslandPopL)&gt;0],
       text.col=glob.popsL.pal[levels(svglob.D$PopulationL)[table(svglob.D$IslandPopL)&gt;0]], 
       cex =1.2, inset=c(-0.03,0), xpd=T, bty=&quot;n&quot;)
legend(&quot;bottomright&quot;, c(&quot;Female&quot;,&quot;Herm.&quot;,&quot;Male&quot;), cex =1.2, pch = c(1,5,0), pt.cex=1.5, inset=c(-0,0), xpd=T, bty=&quot;n&quot;, pt.lwd=4)
legend(&quot;topleft&quot;, &quot;Day&quot;, bty=&quot;n&quot;, cex=1.2, inset=c(-0.04, 0))
dev.copy(cairo_pdf, filename=&quot;./output_cut/AppendixS3_CAP_C.pdf&quot;, width=8, height=8)
dev.off()
}</code></pre>
<p><img src="figures2018/cap_Sex_Pop-1.svg" width="768" /></p>
<pre class="r"><code>#S globosa scent distance matrix vs. GEOGRAPHIC distance mantel test

#get pops from kml
pops &lt;- st_read(&quot;./data/Schiedea populations.kml&quot;, quiet=T) 
#&quot;http://www.google.com/maps/d/u/0/kml?forcekml=1&amp;mid=1nrFQN_CJTffw2Smd3-dD9C-sbqU&amp;lid=2WoGNCc9HTw&quot;) #From Google My Map
pops &lt;- cbind(st_coordinates(pops), pops)
rownames(pops) &lt;- pops$Pop_Number

#distance with latlong
latlong.glob &lt;- as.data.frame(pops[match(svglob$Population, pops$Pop_Number), ])[,c(&quot;X&quot;,&quot;Y&quot;)]#make a dataframe of latlong coordinates with the same rows as vol.glob
rownames(latlong.glob) &lt;- rownames(svglob)
geogdist.glob &lt;- dist(latlong.glob)

#actual distance in meters
geogdist.glob.m &lt;-  st_distance(pops[match(svglob$Population, pops$Pop_Number), ])

scentdist.glob &lt;- vegdist(sqrt(fvol.glob), method=&quot;bray&quot;)
mantel.geog &lt;- mantel(scentdist.glob, geogdist.glob.m)

scent.geog.glob &lt;- data.frame(scent=as.vector(scentdist.glob), geog=as.vector(as.dist(geogdist.glob.m)))

#AVERAGE scent distance vs. geoographic distance
#just one distance per pair of populations (all 12 populations)
#actual distance in meters
geog.m &lt;- st_distance(pops[match(levels(svglob$Population), pops$Pop_Number), ])
rownames(geog.m) &lt;- colnames(geog.m) &lt;- levels(svglob$Population)
geog.glob.m &lt;- geog.m[levels(svglob$Population),levels(svglob$Population)]

mean12.scent &lt;- meandist(scentdist.glob, svglob$Population)
mantel.avggeog &lt;- mantel(mean12.scent, geog.glob.m)</code></pre>
<pre class="r"><code>#geographic matricies for genetics
#distance with latlong
mean.geog &lt;- dist(as.data.frame(pops[match(levels(svglob$Population), pops$Pop_Number), ])[,c(&quot;X&quot;,&quot;Y&quot;)])
mean.geog.glob &lt;- as.matrix(mean.geog)[levels(svglob$PopulationL),levels(svglob$PopulationL)]

#actual distance in meters
mean.geog.m &lt;- st_distance(pops[match(levels(svglob$PopulationL), pops$Pop_Number), ])
rownames(mean.geog.m) &lt;- colnames(mean.geog.m) &lt;- levels(svglob$PopulationL)
mean.geog.glob.m &lt;- mean.geog.m[levels(svglob$PopulationL),levels(svglob$PopulationL)]

#S globosa scent distance matrix vs. GENETIC distance mantel test
mean.scent &lt;- meandist(scentdist.glob, svglob$PopulationL)

mean.gen &lt;- lapply(meand, function(dm) as.matrix(dm)[levels(svglob$PopulationL),levels(svglob$PopulationL)])
mean.avggen &lt;- as.matrix(avgd)[levels(svglob$PopulationL),levels(svglob$PopulationL)]

mean.avggen.labels &lt;- expand.grid(rownames(mean.avggen),rownames(mean.avggen))
mean.avggen.labels &lt;- mean.avggen.labels %&gt;% mutate(dupe = duplicated(t(apply(mean.avggen.labels, 1, sort))),
                                      intrapop = Var1==Var2,
                                      combo = paste(Var1, Var2),
                                      Island1 = pops[as.character(Var1),]$Island,
                                      Island2 = pops[as.character(Var2),]$Island,
                                      Kanehoalani = (Var1==&quot;964&quot;|Var2==&quot;964&quot;),
                                      Hawaii = Island1 ==&quot;Hawai&#39;i&quot; | Island2==&quot;Hawai&#39;i&quot;,
                                      Group = c(&quot;O&#39;ahu &amp; Maui Nui&quot;, &quot;Hawai&#39;i&quot;, &quot;O&#39;ahu 964&quot;)[as.integer(factor(paste(ifelse(Kanehoalani,Kanehoalani,Hawaii), Kanehoalani)))])

mantel.gen &lt;- lapply(mean.gen, mantel, mean.scent)
mantel.avggen &lt;- mantel(mean.avggen, mean.scent)</code></pre>
<pre class="r"><code>#GEOG and GENETICS: Split by time and sex
geogdist.glob.m.split &lt;- scentdist.glob.split &lt;- mantel.geog.split &lt;- lm.geog.split &lt;- mean.scent.split &lt;- mean.geogdist.glob.m.split &lt;-mantel.gen.split &lt;-mean.gen.split &lt;-mean.avggen.split &lt;-  mantel.avggen.split &lt;- lm.avggen.split &lt;- mean12.scent.split &lt;- geog12dist.glob.m.split &lt;- mantel.geog12.split &lt;- lm.geog12.split &lt;- mantel.partial.avggen.split &lt;- mantel.partial.geog.split &lt;- list()
for(t in levels(svglob$DN)){
  t1 &lt;- substr(t,1,1)
  for(s in levels(svglob$Sex)){
    #geogdist.glob.split[[t1]][[s]] &lt;- dist(latlong.glob[svglob$DN==t &amp; svglob$Sex==s,])
    geogdist.glob.m.split[[t1]][[s]] &lt;- as.dist(st_distance(pops[match(svglob$Population[svglob$DN==t &amp; svglob$Sex==s], pops$Pop_Number), ]))
    scentdist.glob.split[[t1]][[s]] &lt;- vegdist(decostand(vol.glob[svglob$DN==t &amp; svglob$Sex==s,], &quot;hellinger&quot;), method=&quot;bray&quot;)
    mean.scent.split[[t1]][[s]] &lt;- meandist(scentdist.glob.split[[t1]][[s]], svglob[svglob$DN==t &amp; svglob$Sex==s,]$PopulationL)
    mean.geogdist.glob.m.split[[t1]][[s]] &lt;- st_distance(pops[match(rownames(mean.scent.split[[t1]][[s]]), pops$Pop_Number), ])
    #diag(mean.scent.split[[t1]][[s]]) &lt;- NA #NEW, this takes out the within-population scent distances from plotting
    mantel.geog.split[[t1]][[s]] &lt;- mantel(scentdist.glob.split[[t1]][[s]], geogdist.glob.m.split[[t1]][[s]])
    lm.geog.split[[t1]][[s]] &lt;- tryCatch(lm(as.vector(scentdist.glob.split[[t1]][[s]]) ~ as.vector(geogdist.glob.m.split[[t1]][[s]])), error=function(e) NA)
    pops.t.s &lt;- rownames(mean.scent.split[[t1]][[s]])
    
    #scent ~ geographic distance averaged by population for all 12 pops (not lumped)
    mean12.scent.split[[t1]][[s]] &lt;- meandist(scentdist.glob.split[[t1]][[s]], svglob[svglob$DN==t &amp; svglob$Sex==s,]$Population)
    pops12.t.s &lt;- rownames(mean12.scent.split[[t1]][[s]])
    geog12dist.glob.m.split[[t1]][[s]] &lt;- st_distance(pops[match(pops12.t.s, pops$Pop_Number), ])
    mantel.geog12.split[[t1]][[s]] &lt;- mantel(mean12.scent.split[[t1]][[s]], geog12dist.glob.m.split[[t1]][[s]])
    lm.geog12.split[[t1]][[s]] &lt;- tryCatch(lm(as.vector(mean12.scent.split[[t1]][[s]]) ~ as.vector(geog12dist.glob.m.split[[t1]][[s]])), error=function(e) NA)
    
    for(g in genes) {
       mean.gen.split[[g]][[t1]][[s]] &lt;- mean.gen[[g]][pops.t.s,pops.t.s]
       mantel.gen.split[[g]][[t1]][[s]] &lt;- mantel(mean.scent.split[[t1]][[s]], mean.gen[[g]][pops.t.s,pops.t.s])
    }
    mean.avggen.split[[t1]][[s]] &lt;- mean.avggen[pops.t.s,pops.t.s]
    diag(mean.avggen.split[[t1]][[s]]) &lt;- NA #NEW, this takes out the within-population genetic distances from plotting
    mantel.avggen.split[[t1]][[s]] &lt;- mantel(mean.scent.split[[t1]][[s]], mean.avggen[pops.t.s,pops.t.s])
    lm.avggen.split[[t1]][[s]] &lt;- tryCatch(lm(as.vector(mean.scent.split[[t1]][[s]]) ~ as.vector(mean.avggen[pops.t.s,pops.t.s])), error=function(e) NA)
    
    #Partial Mantel
    mantel.partial.avggen.split[[t1]][[s]] &lt;- mantel.partial(mean.scent.split[[t1]][[s]], mean.avggen[pops.t.s,pops.t.s], mean.geogdist.glob.m.split[[t1]][[s]])
    mantel.partial.geog.split[[t1]][[s]] &lt;- mantel.partial(mean.scent.split[[t1]][[s]], mean.geogdist.glob.m.split[[t1]][[s]], mean.avggen[pops.t.s,pops.t.s])
  }
}

geogdist.glob.df &lt;- data.frame(geog=unlist(geogdist.glob.m.split))
geogdist.glob.df$DN &lt;- factor(substr(rownames(geogdist.glob.df),1,1))
geogdist.glob.df$Sex &lt;- factor(substr(rownames(geogdist.glob.df),3,3))
scent.geog.glob.split &lt;- data.frame(scent=unlist(scentdist.glob.split), geogdist.glob.df)

scentdist.glob.df &lt;- data.frame(scent=unlist(mean.scent.split))
scentdist.glob.df$DN &lt;- factor(substr(rownames(scentdist.glob.df),1,1))
scentdist.glob.df$Sex &lt;- factor(substr(rownames(scentdist.glob.df),3,3))

geog12dist.glob.df &lt;- data.frame(geog=unlist(geog12dist.glob.m.split))
geog12dist.glob.df$DN &lt;- factor(substr(rownames(geog12dist.glob.df),1,1))
geog12dist.glob.df$Sex &lt;- factor(substr(rownames(geog12dist.glob.df),3,3))
scent.geog12.glob.split &lt;- data.frame(scent=unlist(mean12.scent.split), geog12dist.glob.df)
 
scent.gen.glob.split &lt;- data.frame(lapply(mean.gen.split, unlist), scentdist.glob.df)
scent.glob.gen.split.long &lt;- gather(scent.gen.glob.split, &quot;gene&quot;, &quot;gendist&quot;, 1:3, factor_key=T)
#this old commented code used the average of the mean genetic distances (per population) scaled 0-1 within each gene (mean of means).
#scent.avggen.glob.split &lt;- scent.gen.glob.split %&gt;% mutate_at(vars(ncpGS:psbM.trnD), function(x) x/max(x, na.rm=T)) %&gt;% mutate(avggen=rowMeans(cbind(ncpGS,pepC,psbM.trnD), na.rm=T))

#better to take the average scaled genetic distances between individuals, then take the mean by population.
scent.avggen.glob.split &lt;- data.frame(avggen=unlist(mean.avggen.split), scentdist.glob.df)

geog.gen.glob &lt;- data.frame(lapply(mean.gen, as.vector), geog=rep(as.vector(mean.geog.glob.m),3))
geog.gen.globsplit.long &lt;- gather(geog.gen.glob, &quot;gene&quot;, &quot;gendist&quot;, 1:3, factor_key=T)
#geog.avggen.glob &lt;- geog.gen.glob %&gt;% mutate_at(vars(ncpGS:psbM.trnD), function(x) x/max(x, na.rm=T)) %&gt;% mutate(avggen=rowMeans(cbind(ncpGS,pepC,psbM.trnD), na.rm=T))
geog.avggen.glob &lt;- data.frame(avggen=as.vector(mean.avggen), geog=as.vector(mean.geog.glob.m), mean.avggen.labels)

mantel.geog.gen &lt;- lapply(mean.gen, mantel, mean.geog.glob.m)
mantel.geog.avggen &lt;- mantel(mean.avggen, mean.geog.glob.m)

pops.include &lt;- setdiff(glob.pops.WtoE[-c(1,2)],c(&quot;964&quot;,&quot;11221&quot;,&quot;15191&quot;))
mantel.geog.avggen.OMN &lt;- mantel(mean.avggen[pops.include,pops.include], mean.geog.glob.m[pops.include,pops.include])

#NAs in these distances are from 964 or 15191 (1 sample of genetic or scent data each)</code></pre>
<pre class="r"><code>mantels.geog &lt;- mantel.geog12.split %&gt;% unlist(recursive=F) %&gt;%
  sapply(`[`, c(&quot;signif&quot;, &quot;statistic&quot;)) %&gt;% t %&gt;% as.data.frame  %&gt;% 
  tibble::rownames_to_column(&quot;id&quot;) %&gt;% separate(id, c(&quot;DN&quot;,&quot;Sex&quot;), sep=&quot;\\.&quot;) %&gt;% mutate_all(unlist)
lm.geog &lt;- lm.geog12.split %&gt;% unlist(recursive=F) %&gt;%
  lapply(function(x) x$coefficients) %&gt;% as.data.frame %&gt;% t %&gt;% as.data.frame %&gt;% 
  tibble::rownames_to_column(&quot;id&quot;) %&gt;% separate(id, c(&quot;DN&quot;,&quot;Sex&quot;), sep=&quot;\\.&quot;) %&gt;% setNames(c(&quot;DN&quot;,&quot;Sex&quot;,&quot;intercept&quot;, &quot;slope&quot;)) %&gt;% mutate(slope=slope*1000*100) #convert slopes to bray per 100 km (originally in m)
left_join(mantels.geog, lm.geog) %&gt;% filter(Sex !=&quot;H&quot;) %&gt;% 
  kable(caption = &quot;Mantel test of scent vs. geographic distance&quot;, digits=3)</code></pre>
<pre class="r"><code>mantels.avggen &lt;- mantel.avggen.split %&gt;% unlist(recursive=F) %&gt;%
  sapply(`[`, c(&quot;signif&quot;, &quot;statistic&quot;)) %&gt;% t %&gt;% as.data.frame  %&gt;% 
  tibble::rownames_to_column(&quot;id&quot;) %&gt;% separate(id, c(&quot;DN&quot;,&quot;Sex&quot;), sep=&quot;\\.&quot;) %&gt;% mutate_all(unlist)
lm.avggen &lt;- lm.avggen.split %&gt;% unlist(recursive=F) %&gt;%
  lapply(function(x) x$coefficients) %&gt;% as.data.frame %&gt;% t %&gt;% as.data.frame %&gt;% 
  tibble::rownames_to_column(&quot;id&quot;) %&gt;% separate(id, c(&quot;DN&quot;,&quot;Sex&quot;), sep=&quot;\\.&quot;) %&gt;% setNames(c(&quot;DN&quot;,&quot;Sex&quot;,&quot;intercept&quot;, &quot;slope&quot;)) 
left_join(mantels.avggen,lm.avggen) %&gt;% filter(Sex !=&quot;H&quot;) %&gt;% 
  kable(caption = &quot;Mantel test of scent vs. genetic distance&quot;, digits=3)</code></pre>
<pre class="r"><code>mantels.partial.avggen &lt;- mantel.partial.avggen.split %&gt;% unlist(recursive=F) %&gt;%
  sapply(`[`, c(&quot;signif&quot;, &quot;statistic&quot;)) %&gt;% t %&gt;% as.data.frame  %&gt;% 
  tibble::rownames_to_column(&quot;id&quot;) %&gt;% separate(id, c(&quot;DN&quot;,&quot;Sex&quot;), sep=&quot;\\.&quot;) %&gt;% mutate_all(unlist)
mantels.partial.avggen %&gt;% filter(Sex !=&quot;H&quot;) %&gt;% 
  kable(caption = &quot;Partial Mantel test of scent vs. genetic distance, controlling for geographic distance&quot;, digits=3)

mantels.partial.geog &lt;- mantel.partial.geog.split %&gt;% unlist(recursive=F) %&gt;%
  sapply(`[`, c(&quot;signif&quot;, &quot;statistic&quot;)) %&gt;% t %&gt;% as.data.frame  %&gt;% 
  tibble::rownames_to_column(&quot;id&quot;) %&gt;% separate(id, c(&quot;DN&quot;,&quot;Sex&quot;), sep=&quot;\\.&quot;) %&gt;% mutate_all(unlist)
mantels.partial.geog %&gt;% filter(Sex !=&quot;H&quot;) %&gt;% 
    kable(caption = &quot;Partial Mantel test of scent vs. geographic distance, controlling for genetic distance&quot;, digits=3)</code></pre>
<pre class="r"><code>(scent.avggen.plot &lt;- ggplot(scent.avggen.glob.split %&gt;% dplyr::filter(Sex!=&quot;H&quot;), aes(avggen, scent, color=Sex, shape=factor(paste(Sex,DN)), linetype=DN)) +
    geom_point() + 
    #geom_smooth(method=&quot;loess&quot;, se=F, span=3) +
   geom_smooth(method=&quot;lm&quot;, se=F)+
    scale_x_continuous(&quot;Genetic distance&quot;, limits=c(0,NA), expand=expansion(mult=c(0,0.05))) +
    scale_y_continuous(&quot;Scent composition distance&quot;, limits=c(0,NA), expand=expansion(mult=c(0,0.05))) +
    scale_linetype_manual(&quot;Time&quot;, values=c(2,1), labels=c(&quot;Day&quot;, &quot;Night&quot;))+
    scale_shape_manual(&quot;Sex and Time&quot;, values=c(1,16,0,15), labels=c(&quot;Female - Day&quot;,&quot;Female - Night&quot;,&quot;Male - Day&quot;,&quot;Male - Night&quot;))+
    scale_color_brewer(&quot;Sex&quot;, palette=&quot;Set1&quot;, labels=c(&quot;Female&quot;,&quot;Male&quot;))+
    theme_classic() + theme(axis.text=element_text(color=&quot;black&quot;))+
   guides(linetype=guide_legend(keywidth=2, override.aes = list(color=&quot;black&quot;, size=1))))
ggsave(&quot;./output_cut/Fig4split.pdf&quot;, scent.avggen.plot, device=cairo_pdf, height=6, width=7)</code></pre>
<pre class="r"><code>scent.geog.between.plot &lt;- ggplot(scent.geog12.glob.split %&gt;% filter(Sex!=&quot;H&quot;, geog!=0), aes(geog/1000, scent, color=Sex, shape=factor(paste(Sex,DN)), linetype=DN)) + 
   geom_vline(aes(xintercept=55), color=&quot;black&quot;, linetype=3)+
   annotate(&quot;text&quot;, label=&quot;Within\nisland&quot;, x=25, y=0.05) +
   annotate(&quot;text&quot;, label=&quot;Between\nislands&quot;, x=87, y=0.05) +
    geom_point(alpha=1) + 
    geom_smooth(method=&quot;loess&quot;, se=F, span=0.6) +
    scale_x_continuous(&quot;Geographic distance (km)&quot;, limits=c(0,300), expand=expansion(mult=c(0,0)), breaks=seq(0,300, by=50)) +
    scale_y_continuous(&quot;Scent composition distance&quot;, limits=c(0,0.87), expand=expansion(mult=c(0,0.05))) +
    scale_linetype_manual(&quot;Time&quot;, values=c(2,1), labels=c(&quot;Day&quot;, &quot;Night&quot;))+
    scale_shape_manual(&quot;Sex and Time&quot;, values=c(1,16,0,15), labels=c(&quot;Female - Day&quot;,&quot;Female - Night&quot;,&quot;Male - Day&quot;,&quot;Male - Night&quot;))+
    scale_color_brewer(&quot;Sex&quot;, palette=&quot;Set1&quot;, labels=c(&quot;Female&quot;,&quot;Male&quot;))+
    theme_classic() + theme( axis.text=element_text(color=&quot;black&quot;), axis.text.y=element_blank(), axis.title.y=element_blank(), axis.ticks.y=element_blank())+
   guides(linetype=guide_legend(keywidth=2, override.aes = list(color=&quot;black&quot;, size=1))) #need to combine legend

scent.within.plot &lt;- ggplot(scent.geog.glob.split %&gt;% filter(Sex!=&quot;H&quot;, geog==0), aes(x=factor(paste(Sex,DN)), y=scent, color=Sex, shape=factor(paste(Sex,DN)), linetype=DN)) +
 #geom_boxplot(linetype=1, outlier.shape = NA, show.legend = FALSE)+
       stat_summary(fun=mean, geom=&quot;point&quot;, shape=&quot;-&quot;, size=15) +
   geom_point()+
   annotate(&quot;text&quot;, label=&quot;Within\npopulation&quot;, x=2.5, y=0.05) +
      scale_shape_manual(&quot;Sex and Time&quot;, values=c(1,16,0,15), labels=c(&quot;Female - Day&quot;,&quot;Female - Night&quot;,&quot;Male - Day&quot;,&quot;Male - Night&quot;))+
    scale_color_brewer(&quot;Sex&quot;, palette=&quot;Set1&quot;, labels=c(&quot;Female&quot;,&quot;Male&quot;)) + theme_classic() + theme( axis.text=element_text(color=&quot;black&quot;), axis.text.x=element_text(color=&quot;white&quot;), axis.ticks.x=element_blank()) + guides(color=F, shape=F) +
      scale_y_continuous(&quot;Scent composition distance&quot;, limits=c(0,0.87), expand=expansion(mult=c(0,0.05))) +
  scale_x_discrete(&quot;&quot;) 

scent.geog.split.plot &lt;- grid.arrange(scent.within.plot, scent.geog.between.plot, nrow = 1, widths=c(1,4))
ggsave(&quot;./output_cut/Fig5split.pdf&quot;, scent.geog.split.plot, device=cairo_pdf, height=6, width=7)</code></pre>
<pre class="r"><code>if(params$year==2019) {
fvol.glob.mean %&gt;% select(c(13,14,1:7,16,18,23:25)) %&gt;% arrange(type, subtype, desc(freq.glob), name) %&gt;% 
  filter(freq.glob &gt; 0.10) %&gt;% #Present in 10% of samples 
    select(-c(standard,RT, CAS), prop=freq.glob, `ret. index`=RI) %&gt;% 
    kable(caption=&quot;Emissions of each volatile (ng/h/flower)&quot;, digits=2)
}</code></pre>
<pre class="r"><code>mincount &lt;- 15
srt &lt;- order(svglob$DN, svglob$Sex, svglob$Year, svglob$Population,  decreasing=F)
filter.cols &lt;- colSums(decostand(vol.glob, &quot;pa&quot;))&gt;mincount
fvol.glob.filter &lt;- fvol.glob[,filter.cols]

ph &lt;- pheatmap(as.matrix(t(fvol.glob.filter[srt,]))^(1/3),
         cluster_cols=F, show_colnames=F,
         clustering_method=&quot;mcquitty&quot;, clustering_distance_rows=&quot;correlation&quot;,
         clustering_callback = function(hc, ...){dendsort(hc, type=&quot;average&quot;)},
         scale=&quot;none&quot;, color=inferno(512),
         annotation_row = data.frame(Class=factor(fvol.glob.mean$type[filter.cols]), row.names=colnames(fvol.glob.filter)),
         annotation_col = data.frame(Population=svglob$Population, Sex=recode(svglob$Sex, F=&quot;Female&quot;, H=&quot;Herm.&quot;, M=&quot;Male&quot;), Time=svglob$DN, row.names=rownames(fvol.glob.filter)), 
         annotation_colors = list(Class=chemtype.pal,
                                  Time=setNames(c(&quot;grey80&quot;,&quot;grey20&quot;),levels(svglob$DN)),
                                  Sex=setNames(brewer.pal(3,&quot;Set1&quot;)[c(1,3,2)],
                                               levels(recode(svglob$Sex, F=&quot;Female&quot;, H=&quot;Herm.&quot;, M=&quot;Male&quot;))),
                                  Population=glob.pops.pal), 
         gaps_col = which(as.logical(diff(as.integer(factor(paste(svglob$Sex,svglob$DN)))[srt]))),
         cellwidth = 3.5, cellheight = 10, fontsize = 10, border_color = NA, annotation_legend=T, cutree_rows=10,
         legend=T, legend_breaks=seq(0, max(fvol.glob.filter), by= 20)^(1/3), 
         legend_labels=c(&quot;0 ng/h/flower&quot;,seq(20, max(fvol.glob.filter), by= 20)))

cairo_pdf(&quot;./output_cut/AppendixS5.pdf&quot;, width=10.6, height=7); ph; dev.off()</code></pre>
<pre class="r"><code>geog.gen.between.plot &lt;- ggplot(geog.avggen.glob %&gt;% filter(geog!=0), aes(geog/1000, avggen, color=Group)) +
    geom_vline(aes(xintercept=55), color=&quot;black&quot;, linetype=3)+
   annotate(&quot;text&quot;, label=&quot;Within\nisland&quot;, x=25, y=0.02) +
   annotate(&quot;text&quot;, label=&quot;Between\nislands&quot;, x=87, y=0.02) +
    geom_point(shape=19) + 
    geom_smooth(method=&quot;loess&quot;, se=F, span=1.5) +
   scale_color_manual(&quot;Pairwise comparisons&quot;, breaks=c(&quot;O&#39;ahu 964&quot;, &quot;Hawai&#39;i&quot;, &quot;O&#39;ahu &amp; Maui Nui&quot;), 
                      labels=c(&quot;O&#39;ahu 964 vs. all&quot;, &quot;Hawai&#39;i Island vs. all&quot;, &quot;Remaining populations&quot;),
                      values=unname(c(glob.pops.pal[2],glob.pops.pal[12],&quot;black&quot;)), )+
    scale_x_continuous(&quot;Geographic distance (km)&quot;, limits=c(0,300), expand=expansion(mult=c(0,0)), breaks=seq(0,300, by=50), minor_breaks = seq(0,300, by=50)) +
    scale_y_continuous(&quot;Genetic distance&quot;, limits=c(0,0.23), expand=expansion(mult=c(0,0.05)))+
   guides(shape=F)+
    theme_classic()+theme(legend.position = &quot;right&quot;, plot.margin=theme_classic()$plot.margin*1.5, axis.text=element_text(color=&quot;black&quot;), axis.text.y=element_blank(), axis.title.y=element_blank(), axis.ticks.y=element_blank())

geog.gen.within.plot &lt;- ggplot(geog.avggen.glob %&gt;% filter(geog==0), aes(x=0, avggen)) +
    geom_point(shape=1)+
    annotate(&quot;text&quot;, label=&quot;Within\npopulation&quot;, x=-0.75, y=0.02) +
   geom_text_repel(aes(label=ifelse(dupe | !intrapop, &quot;&quot;, as.character(Var1))), size=3, direction=&quot;y&quot;, hjust=0, position=position_nudge(x=-1), box.padding=0, point.padding=0, label.padding=0, show.legend = F)+
    scale_y_continuous(&quot;Genetic distance&quot;, limits=c(0,0.23), expand=expansion(mult=c(0,0.05)))+
    scale_x_continuous(&quot;&quot;, limits=c(-1.5,0)) +
        theme_classic()+ theme(plot.margin=theme_classic()$plot.margin*1.5, axis.text=element_text(color=&quot;black&quot;), axis.text.x=element_text(color=&quot;white&quot;), axis.ticks.x=element_blank())

geog.gen.split.plot &lt;- grid.arrange(geog.gen.within.plot, geog.gen.between.plot, nrow = 1, widths=c(1,3.6))
ggsave(&quot;./output_cut/AppendixS6.pdf&quot;, geog.gen.split.plot, device=cairo_pdf, height=6, width=7)</code></pre>
<pre class="r"><code>print(&quot;Mantel test of genetic vs. geographic distance&quot;)
mantel.geog.avggen
print(&quot;Mantel test of genetic vs. geographic distance, excluding Hawai&#39;i Island&quot;)
mantel.geog.avggen.OMN</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
